src/ixdat/data_series.py

```diff
@@ -0,0 +1,216 @@
+"""This module defines the DataSeries class, the elementary data structure of ixdat
+
+An ixdat DataSeries is a wapper around a numpy array containing the metadata needed
+to combine it with other DataSeries. Typically this means a reference to the time
+variable corresponding to the rows of the array. The time variable itself is a special
+case, TimeSeries, which must know its absolute (unix) time.
+"""
+
+from .db import Saveable
+from .units import Unit
+from .exceptions import TimeError, AxisError
+
+
+class DataSeries(Saveable):
+    """The base class for all numerical data representation in ixdat.
+
+    These class's objects are saved and loaded as rows in the data_series table
+    """
+
+    table_name = "data_series"
+    column_attrs = {"id": "i", "name": "name", "unit": "unit_name", "data": "data"}
+
+    def __init__(self, name, unit_name, data):
+        """initialize a data series with its name, unit, and data (id handled by parent)
+
+        Args:
+            name (str): The name of the data series
+            unit_name (str): The name of unit in which the data is stored
+            data (np.array): The numerical data
+        """
+        super().__init__()
+        self.name = name
+        self.unit = Unit(unit_name)
+        self._data = data
+
+    @classmethod
+    def from_dict(cls, obj_as_dict):
+        """Return the right type of DataSeries based on the info in its serialization"""
+        if "tstamp" in obj_as_dict:
+            return TimeSeries(**obj_as_dict)
+        elif "t_id" in obj_as_dict:
+            return ValueSeries(**obj_as_dict)
+        elif "a_ids" in obj_as_dict:
+            return Field(**obj_as_dict)
+
+    def __repr__(self):
+        return f"{self.__class__}(id={self.id}, name='{self.name}')"
+
+    @property
+    def data(self):
+        """The data as a np.array, loading lazily."""
+        if self._data is None:
+            self._data = self.load_data()  # inherited from Saveable.
+        return self._data
+
+    @property
+    def unit_name(self):
+        """The name of the data series' unit"""
+        return self.unit.name
+
+
+class TimeSeries(DataSeries):
```

Comment:
I think `DataSeries.__repr__` as it's written will show the inheriting class name? The only arguments are id and name, which are universal to all of them. Usually the name will make it clear what it is. `__repr__` could be added to inheriting classes to show more arguments like `tstamp` but I'm not sure this really helps the user anyway (unix timestamps aren't super readable).

Reply:
Confident enough about my answer to mark this resolved without a TODO - will be a clear fix if it doesn't work as expected.

Reply:
It will show the inheriting class name if you use `self.__class__` in the repr string, so that is fine. I commented only because I though I saw a subclass with a different signature. I was probably mistaken :D